File: python/breakout/bar.py
import pygame


class Bar:
    def __init__(self, screen, settings):
        self.screen = screen
        self.screen_rect = self.screen.get_rect()
        self.settings = settings

        # bar dimensions
        self.width = 200
        self.height = 30

        # bar color
        self.color = (255, 255, 255)

        # rect
        self.rect = pygame.Rect(0, 0, self.width, self.height)
        self.rect.center = self.screen_rect.center
        self.rect.bottom = self.screen_rect.height

        # flags
        self.flag_right = False
        self.flag_left = False

    def draw(self):
        if self.flag_left and self.rect.left >= 0 and not self.settings.game_over:
            self.rect.left -= 1

        if self.flag_right and self.rect.right <= self.screen_rect.width and not self.settings.game_over:
            self.rect.right += 1

        pygame.draw.rect(self.screen, self.color, self.rect)

File: java/BreakOut/src/breakout/Bar.java
package breakout;
import java.awt.Color;
import java.awt.Graphics2D;
import java.awt.Rectangle;

public class Bar {
    Settings settings;
    
    //position
    protected final int startx;
    protected int moveValue;
    
    //rectangle
    protected Rectangle rect;
    
    public Bar(Settings set){
        startx = 400;
        moveValue = 20;
        settings = set;
        rect = new Rectangle(startx, 546, 200, 30);
    }
    
    protected void draw(Graphics2D g){
        g.setColor(Color.white);
        g.fillRect(rect.x, rect.y, rect.width, rect.height);
    }
    
    protected void reset(){
        rect.x = startx;
        if(settings.gameOver) moveValue = 20;
    }
    
}
